hazelcast/src/main/java/com/hazelcast/sql/impl/SqlResultImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl;
+
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlRowMetadata;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.state.QueryInitiatorState;
+import com.hazelcast.sql.impl.state.QueryState;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Cursor implementation.
+ */
+public class SqlResultImpl implements SqlResult {
```

Comment:
no `toString` implementation for this class, is that intentional?

Reply:
You mean to produce some string view of the whole result? We can't iterate the iterator here because it's a mutating operation. Years ago I had an issue that displaying a variable in the debugger called the `toString` and it lazy-loaded the value and then I couldn't reproduce my issue. If we want to format the results somehow, it should be a special method where we could document it, not `toString`.

Reply:
There is nothing to expose here. Row and row metadata do have `toString` implementations.